### DISCUSSION TITLE
[QUESTION]:ilogtail输出到kafka，默认配置情况下出现消息过大被kafka拒绝

### DISCUSSION THREAD

**Discussion by hailangxing** [alibaba/ilogtail#1555] [Help] [answered]
![image](https://github.com/alibaba/ilogtail/assets/8348166/1383bf1b-f350-4a4e-b19d-5379ed7a1d8c)

![image](https://github.com/alibaba/ilogtail/assets/8348166/0f2f57ab-24d7-48cd-83e2-f31d5785f168)

使用默认的MaxMessageBytes配置，会出现kafka Message was to large,service rejected it to avoid allocation error，从MaxMessageBytes上看是0.96m，没有超过kafka默认消息大小限制1m

**Comment by shalousun** [comment, marked as answer]
@hailangxing 这种kafka的客户端都是根据服务端状态返回状态的错误了，直接把服务端和客户端的都提升就OK了。本质上一条日志采集上来，发送到kafka最终的message会附带一些信息会比原来的日志大。这个也不属于是iLogtail的问题，发送消息都会碰到。设置留足空余，极限值很容易被拒绝。通常建议是broker端的配置要比producer端高。
